mmengine/registry/build_functions.py

```diff
@@ -120,10 +121,15 @@ def build_from_cfg(
             else:
                 obj = obj_cls(**args)  # type: ignore
 
+            # For some rare cases (e.g. obj_cls is a partial function), obj_cls
+            # doesn't have the following attributes. Use default value to
+            # prevent error
+            cls_name = getattr(obj_cls, '__name__', str(obj_cls))
+            cls_module = getattr(obj_cls, '__module__', 'unknown')
```

Comment:
Can we use `isinstance(obj_cls, functools.partial)` to check whether it is a partial function? If it is a partial function, we can get its name and module from `obj_cls.func`.

Reply:
Before this PR, `Registry` only support functions and classes, whose definition/implementation can all be found in their `__module__`. This is good for debugging.

However, this PR allows registering generic callable `object`, whose debug info should be where/how they are created. As far as I know, this info seems only accessible for users who created this `object`.

Reply:
For partial functions, `str(obj_cls)` seems to contain enough debug information.
```python
>>> def foo(x):
...     pass
... 
>>> bar=functools.partial(foo, x=0)
>>> bar
functools.partial(<function foo at 0x7f8fd908a7a0>, x=0)
>>> str(bar)
'functools.partial(<function foo at 0x7f8fd908a7a0>, x=0)'
```

Reply:
Got it